modules/ml/src/main/java/org/apache/ignite/ml/composition/BaggingModelTrainer.java

```diff
@@ -89,11 +91,26 @@ public BaggingModelTrainer(PredictionsAggregator predictionsAggregator,
         IgniteBiFunction<K, V, Vector> featureExtractor,
         IgniteBiFunction<K, V, Double> lbExtractor) {
 
-        List<ModelOnFeaturesSubspace> learnedModels = new ArrayList<>();
-        for (int i = 0; i < ensembleSize; i++)
-            learnedModels.add(learnModel(datasetBuilder, featureExtractor, lbExtractor));
-
-        return new ModelsComposition(learnedModels, predictionsAggregator);
+        MLLogger log = environment.logger(getClass());
+        log.log(MLLogger.VerboseLevel.MIN, "Start learning");
+
+        Long startTs = System.currentTimeMillis();
+        List<Promise<ModelOnFeaturesSubspace>> learnedModelsF = new ArrayList<>();
+        for (int i = 0; i < ensembleSize; i++) {
+            learnedModelsF.add(environment.parallelismStrategy().submit(() ->
```

Comment:
Strange part of code, need a time to understand what's happened here

Reply:
I have implemented additional default method for Promise that uses list of suppliers
but it doesn't looks better